llvm/include/llvm/SYCLLowerIR/ModuleSplitter.h

```diff
@@ -18,6 +18,7 @@
 #include "llvm/ADT/SetVector.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
```

Comment:
nit: do we need this instead of forward declaring?

Reply:
We do. Unlike `shared_ptr`, `unique_ptr` insists on the contained type not being incomplete.

Reply:
But why do we need this in a header file which is otherwise unchanged? It seems like either this `#include` should be moved straight into `.cpp` file, or into another `.h` file

Reply:
It seems to be a dependency bug that just wasn't noticed because `Module.h` was included before `ModuleSplitter.h` in all other cases.